File: practice/archives/oct_7/A_Triangle.py

# ⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⣿⣶⣄⣀⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
# ⠀⠀⠀⠀⠀⠀⠀⢀⣴⣿⣿⣿⣿⣿⣿⣿⣿⣿⣶⣦⣄⣀⡀⣠⣾⡇⠀⠀⠀⠀
# ⠀⠀⠀⠀⠀⠀⣴⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡇⠀⠀⠀⠀
# ⠀⠀⠀⠀⢀⣾⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠿⠿⢿⣿⣿⡇⠀⠀⠀⠀
# ⠀⣶⣿⣦⣜⣿⣿⣿⡟⠻⣿⣿⣿⣿⣿⣿⣿⡿⢿⡏⣴⣺⣦⣙⣿⣷⣄⠀⠀⠀
# ⠀⣯⡇⣻⣿⣿⣿⣿⣷⣾⣿⣬⣥⣭⣽⣿⣿⣧⣼⡇⣯⣇⣹⣿⣿⣿⣿⣧⠀⠀
# ⠀⠹⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠸⣿⣿⣿⣿⣿⣿⣿⣷⠀
A = list(map(int,input().strip().split()))

p1 = (A[0],A[1])
p2 = (A[2],A[3])
p3 = (A[4],A[5])

def is_right(a,b,c):
    u = (b[0] - a[0],b[1] - a[1])
    v = (c[0] - a[0],c[1] - a[1])
    return (u[0]*v[0] + u[1]*v[1] == 0) and u != (0,0) and v != (0,0) and u != v 


def is_almost(a,b,c):
    dirn = [(0,1),(0,-1),(1,0),(-1,0)]
    for dx,dy in dirn:
        if is_right((a[0]+dx,a[1]+dy),b,c) or is_right(a,(b[0]+dx,b[1]+dy),c) or is_right(a,b,(c[0]+dx,c[1]+dy)): 
            return True
    return False

def solve(p1,p2,p3):
    for p in [(p1,p2,p3),(p2,p3,p1),(p3,p1,p2)]:
        if is_right(*p): return "RIGHT"
    for p in [(p1,p2,p3),(p2,p3,p1),(p3,p1,p2)]:
        if is_almost(*p): return "ALMOST"
    return "NEITHER"

print(solve(p1,p2,p3))
    

File: practice/archives/oct_7/A_Triangle.cpp
#define _USE_MATH_DEFINES

#include <iostream>
#include <cstdio>
#include <vector>
#include <cmath>
#include <list>
#include <iomanip>
#include <stack>
#include <map>
#include <set>
#include <queue>
#include <string>
#include <algorithm>
#include <cstring>
#include <cstdlib>
#include <cassert>
#include <cstring>
#include <ctime>

#define all(a) a.begin(),a.end()
#define pb push_back
#define mp make_pair
#define forn(i,n) for(int i = 0; i < int(n); ++i)
#define sz(a) int(a.size())

using namespace std;

typedef long long li;
typedef long double ld;

typedef pair<int,int> pt;
#define ft first
#define sc second

int a[6];

bool read() {
	forn(i, 6)
		if (!(cin >> a[i]))
			return false;
	return true;
}

#define sqr(x) ((x)*(x))
int dist2(int x1, int y1, int x2, int y2) {
	return sqr(x1 - x2) + sqr(y1 - y2);
}

bool check() {
	int c[3], csz = 0;

	forn(i, 3)
		forn(j, i)
			c[csz++] = dist2(a[i * 2], a[i * 2 + 1], a[j * 2], a[j * 2 + 1]);

	sort(c, c + 3);
	if (c[0] == 0)
		return false;
	return c[0] + c[1] == c[2];
}

void solve() {
	int res = 0;

	if (check()) {
		puts("RIGHT");
		return;
	}

	for (int d = -1; d <= +1; ++d) {
		forn(i, 6) {
			a[i] -= d;
			if (check()) {
				puts("ALMOST");
                for(int i = 0; i < 6 ; i++) cout << a[i] << ' ';
                cout << '\n'; 
				return;
			}
			a[i] += d;
		}
	}

	puts("NEITHER");
}

int main() {


	while (read())
		solve();
	
	return 0;
}
